File: chat-application-backend-go/controllers/chatController.go
// controllers/chatController.go
package controllers

import (
	"encoding/json"
	"net/http"
	"time"

	"chat-application-backend-go/config"
	"chat-application-backend-go/models"

	_ "github.com/go-sql-driver/mysql"
)

func SendMessage(w http.ResponseWriter, r *http.Request) {
	var message models.Message
	json.NewDecoder(r.Body).Decode(&message)
	message.Timestamp = time.Now()

	db := config.DB // Assuming you have a DB field in your config for MySQL
	_, err := db.Exec("INSERT INTO messages (text, timestamp) VALUES (?, ?)", message.Text, message.Timestamp)
	if err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}

	w.WriteHeader(http.StatusCreated)
	json.NewEncoder(w).Encode(message)
}

func GetMessages(w http.ResponseWriter, r *http.Request) {
	db := config.DB // Assuming you have a DB field in your config for MySQL
	rows, err := db.Query("SELECT text, timestamp FROM messages")
	if err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
	defer rows.Close()

	var messages []models.Message
	for rows.Next() {
		var message models.Message
		if err := rows.Scan(&message.Text, &message.Timestamp); err != nil {
			http.Error(w, err.Error(), http.StatusInternalServerError)
			return
		}
		messages = append(messages, message)
	}

	json.NewEncoder(w).Encode(messages)
}

File: chat-application-backend/controllers/chatController.js
// controllers/chatController.js
const db = require('../config/db'); // Import the database connection

exports.sendMessage = async (req, res) => {
  const { text } = req.body;
  const userId = req.userId; // Get userId from the auth middleware

  try {
    const [result] = await db.execute(
      'INSERT INTO messages (user_id, text) VALUES (?, ?)',
      [userId, text]
    );
    res.status(201).json({ id: result.insertId, userId, text });
  } catch (error) {
    console.error('Error saving message:', error);
    res.status(500).json({ message: 'Error saving message' });
  }
};

exports.getMessages = async (req, res) => {
  try {
    const [rows] = await db.execute('SELECT user_id, text, timestamp FROM messages ORDER BY timestamp ASC');
    res.json(rows);
  } catch (error) {
    console.error('Error retrieving messages:', error);
    res.status(500).json({ message: 'Error retrieving messages' });
  }
};
